addons/base_geolocalize/models/base_geocoder.py

```diff
@@ -98,6 +99,40 @@ def _call_openstreetmap(self, addr, **kw):
         geo = result[0]
         return float(geo['lat']), float(geo['lon'])
 
+    @api.model
+    def _call_openstreetmap_reverse(self, lat, lon):
+        """
+        Use Openstreemap Nominatim service to retrieve location from latitude and longitude
+        :param lat: Latitude
+        :param lon: Longitude
+        :return: Address string or None if not found
+
+        """
+        if not (lat and lon):
+            _logger.info("invalid latitude or longitude given")
+            return None
+        if tools.config['test_enable'] or modules.module.current_test:
+            raise UserError(_("OpenStreetMap calls disabled in testing environment."))
+        try:
+            headers = {"User-Agent": "Odoo (http://www.odoo.com/contactus)"}
+            response = requests.get(
+                "https://nominatim.openstreetmap.org/reverse",
+                headers=headers,
+                params={"format": "json", "lat": lat, "lon": lon},
+                timeout=10,
+            )
```

Comment:
Coucou @odoo/rd-security 

Can we have an override? 

We do an rpc to external service (Open Street Map) to get the localisation according to the latitude and longitude we got. 🙂 

The url used is static and we add a timeout as asking. 🙂 

Thanks in advance!

Have a nice day! 🙂

Reply:
Coucou @odoo/rd-security 

Can we have an override? 

We do an rpc to external service (Open Street Map) to get the localisation according to the latitude and longitude we got. 🙂 

The url used is static and we add a timeout as asking. 🙂 

Thanks in advance!

Have a nice day! 🙂